### ISSUE TITLE
Fuse Mount Option

### ISSUE THREAD

**Issue by dany20mh** [dweidenfeld/plexdrive#8]
Not a big deal, but I couldn’t use any of fuse mount option except the allow_other

Only the last line working.

```
Danials-iMac:~ danial$ ./plexdrive -log-level 2 -temp /Volumes/Backup/plexdrive -fuse-options allow_other,atomic_o_trunc,auto_cache ./mtng2
[PLEXDRIVE] [2017-05-04 07:31] INFO   : Mounting path ./mtng2
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x43a1564]

goroutine 1 [running]:
bazil.org/fuse.Mount(0x7fff5fbff98d, 0x7, 0xc42013b830, 0x6, 0x6, 0xc42013b830, 0x3, 0x6)
	/go/src/bazil.org/fuse/fuse.go:162 +0x114
main.Mount(0xc42013a660, 0x7fff5fbff98d, 0x7, 0xc420197bc0, 0x3, 0x3, 0x0, 0x0)
	/go/src/github.com/dweidenfeld/plexdrive/mount.go:57 +0x784
main.main()
	/go/src/github.com/dweidenfeld/plexdrive/main.go:116 +0xb07



Danials-iMac:~ danial$ ./plexdrive -log-level 2 -temp /Volumes/Backup/plexdrive -fuse-options allow_other,auto_cache ./mtng2
[PLEXDRIVE] [2017-05-04 07:31] INFO   : Mounting path ./mtng2
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x43a1564]

goroutine 1 [running]:
bazil.org/fuse.Mount(0x7fff5fbff98e, 0x7, 0xc42012b800, 0x5, 0x6, 0xc42012b800, 0x3, 0x6)
	/go/src/bazil.org/fuse/fuse.go:162 +0x114
main.Mount(0xc42012a630, 0x7fff5fbff98e, 0x7, 0xc4201ca140, 0x2, 0x2, 0x0, 0x0)
	/go/src/github.com/dweidenfeld/plexdrive/mount.go:57 +0x784
main.main()
	/go/src/github.com/dweidenfeld/plexdrive/main.go:116 +0xb07



Danials-iMac:~ danial$ ./plexdrive -log-level 2 -temp /Volumes/Backup/plexdrive -fuse-options auto_cache ./mtng2
[PLEXDRIVE] [2017-05-04 07:31] INFO   : Mounting path ./mtng2
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x43a1564]

goroutine 1 [running]:
bazil.org/fuse.Mount(0x7fff5fbff98a, 0x7, 0xc42012d710, 0x4, 0x6, 0xc42012d710, 0x3, 0x6)
	/go/src/bazil.org/fuse/fuse.go:162 +0x114
main.Mount(0xc42012c540, 0x7fff5fbff98a, 0x7, 0xc4201ca260, 0x1, 0x1, 0x0, 0x0)
	/go/src/github.com/dweidenfeld/plexdrive/mount.go:57 +0x784
main.main()
	/go/src/github.com/dweidenfeld/plexdrive/main.go:116 +0xb07


Danials-iMac:~ danial$ ./plexdrive -log-level 2 -temp /Volumes/Backup/plexdrive -fuse-options atomic_o_trunc ./mtng2
[PLEXDRIVE] [2017-05-04 07:31] INFO   : Mounting path ./mtng2
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x43a1564]

goroutine 1 [running]:
bazil.org/fuse.Mount(0x7fff5fbff98e, 0x7, 0xc420133770, 0x4, 0x6, 0xc420133770, 0x3, 0x6)
	/go/src/bazil.org/fuse/fuse.go:162 +0x114
main.Mount(0xc4201325a0, 0x7fff5fbff98e, 0x7, 0xc4201d0310, 0x1, 0x1, 0x0, 0x0)
	/go/src/github.com/dweidenfeld/plexdrive/mount.go:57 +0x784
main.main()
	/go/src/github.com/dweidenfeld/plexdrive/main.go:116 +0xb07


Danials-iMac:~ danial$ ./plexdrive -log-level 2 -temp /Volumes/Backup/plexdrive -fuse-options allow_other ./mtng2
```

**Comment by dweidenfeld** [comment]
There was only the need for allow_other.
@dany20mh 
Could you please list the mount options you need and I can try to implement them ;)

**Comment by jaketame** [comment]
max_read=N
              With this option the maximum size of read operations can be set.
              The default is infinite. Note that the size of read requests  is
              limited anyway to 32 pages (which is 128kbyte on i386).

       max_readahead=N
              Set  the  maximum number of bytes to read-ahead.  The default is
              determined by the kernel. On linux-2.6.22 or earlier it's 131072
              (128kbytes)

I've complied a custom kernel that increased the pages to 256 and readahead to 10MB.

You could also add large_read, auto_cache, direct_io,

**Comment by dany20mh** [comment]
@dweidenfeld 
Those are some of them I used too.

@jaketame 
I read some places that large_read is not use anymore and auto_cache is better.

**Comment by dweidenfeld** [comment]
I'm using bazil's fuse module for golang... I don't think it supports all of those options.

The currently supported options are:
 AllowDev() 
 AllowNonEmptyMount() 
 AllowOther() 
 AllowRoot() 
 AllowSUID() 
 AsyncRead() 
 DaemonTimeout(name string) 
 DefaultPermissions() 
 ExclCreate() 
 FSName(name string) 
 LocalVolume() 
 MaxReadahead(n uint32) 
 NoAppleDouble() 
 NoAppleXattr() 
 OSXFUSELocations(paths ...OSXFUSEPaths) 
 ReadOnly() 
 Subtype(fstype string) 
 VolumeName(name string) 
 WritebackCache() 


So enabling the max_readahead is definitely possible, auto_cache, direct_io and max_read is not possible right now.

**Comment by jaketame** [comment]
There is another Go-Fuse package: https://godoc.org/github.com/hanwen/go-fuse/fuse#MountOptions

Makes use of the ones we are after

**Comment by jaketame** [comment]
Looks like direct_io is busted in kernel 4.1 + with a fix in 4.7;
https://sourceforge.net/p/fuse/mailman/message/35276378/
https://patchwork.kernel.org/patch/9346217/

**Comment by headshaker** [comment]
volname=MyName option is not working... And in OS X the default mount name is long as hell...

**Comment by dweidenfeld** [comment]
Those option is also not implemented right now. I will check the other fuse library, but I don't think it will be on a short roadmap ;)

**Comment by dweidenfeld** [comment]
Well... I have had a look on the other fuse implementation. I encountered playback issues with plex using go-fuse. I am not quite sure why. This needs further investigations and will not be tracked in this issue.

Meanwhile, I added some other fuse mount options, like max_readahead etc. All supported options are listed in the README in the master branch. 
Those options will be available in next release.
